src/transformers/models/t5/modeling_t5.py

```diff
@@ -491,7 +491,9 @@ def project(hidden_states, proj_layer, key_value_states, past_key_value):
             # if key and values are already calculated
             # we want only the last query position bias
             if past_key_value is not None:
-                position_bias = position_bias[:, :, -seq_length:, :]
+                # Small optimization for easier export
+                # seq_length == hidden_states.shape[1]
+                position_bias = position_bias[:, :, -hidden_states.shape[1] :, :]
```

Comment:
I'm really surprised this makes onnx life easier, could you give some more detail on why this is?

Reply:
I have just stumbled upon the same bug. Torch ONNX tracer [converts](https://pytorch.org/docs/stable/onnx.html#torch.onnx.operators.shape_as_tensor) `torch.Shape` values from scalars to `torch.Tensor`, which is somehow manipulated by the tracer as the graph is being executed. In this particular case the value of `seq_length` variable, which is now a tensor, changes between the moment it was declared and the moment it was used in the code above.

I'm suspect the change above is using `num_heads` instead of `seq_length`. While it may work, it may not be correct. I suggest to do this instead:

```python
position_bias_seq_length = scores.shape[2]
position_bias = position_bias[:, :, -position_bias_seq_length:, :]
```

because the next line adds the bias to the scores:

```python
scores += position_bias
```

Let me know if I should submit a separate PR.